### DISCUSSION TITLE
Nixie Freaked Out Again--One Week Later

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#986] [Q&A]
Last Friday night/Saturday morning my Nixie seemed to have a nervous breakdown.  I posted my odd experience here https://github.com/tagyoureit/nodejs-poolController/discussions/980.  Eventually, I changed the SD card to a backup unit & did a hard reboot for the umpteenth time, & all seemed to be OK.  And it was until early this morning around 1:20 AM.  Then Nixie had another meltdown...

When I got up this morning around 6:00, I  noticed the pool was running.  I know it shut down last night as scheduled (my filter pump schedule runs the pump from 8:00 AM to 8:00 PM).  So, I fired up dashPanel to see what was happening.

As was  the case last week, the Freeze Protect circuit was on!  The coolest the air here was  last night was around 82 F, hardly a freezing situation.  In addition, the air and pool water temperatures, according to both Nixie & REM were both 0 F.  Finally, flow cell switch was  reporting no water movement, even though the pump was spinning away at 1750 RPM & the Nixie said the flow rate was  18 GPM.  I went to the equipment pad & the flow switch float was at the top of the input side, so the switch should have shown water was moving along nicely in the cell.

REM showed all three inputs to the Meg-Bas were zero, the two temperature readings and the flow switch input from the flow cell.  Happily, this week Nixie did not attempt to drain the entire pool using the overflow drain line!  But, the temperature & flow cell switch readings were wrong & Nixie turned on the pump in response (I assume) to the erroneous readings.

I shutdown the controller (sudo shutdown now) & then killed the breaker.

The readings reported by REM were similar to last week's experience.

Is my Mega-Bas toast?  I'm not sure how to trouble-shoot this situation.  The Mega-Bas is  a version 4.2 unit that's been in use since January.  This  version of the board has DIP switches instead of jumpers to set the inputs & outputs from the board.  I am not aware of any other board changes, but who knows if they changed any other components (I don't have an older unit to compare it with).

Should I reinstall the Mega-Bas software?  Go back to another backup SD card?  Buy a replacement Mega-Bas?  Get a new Pi?  Is  there something evil about weekends that is causing this behavior?

I'm very concerned that the pool will "go wild" again if I can't fix the source of the problem.

Thanks for your help & suggestions!!

P.S., I have a case fan that turns on whenever the Pi CPU temperature exceeds 90F.  This time of year, the fan is always on & the Pi temperature I've seen varies from about 110F to 130F.  The Pi is a 3B+ using Ethernet, not WiFi.  Nixie & REM appear to be using far less than 5% of the CPU, so even if it slows due to temperature, a slower CPU shouldn't be an issue.  I am running the Pi headless so there is no front-end.  I access the Pi via SSH, dashPanel, REM's web page, or Home Assistant.

**Comment by FlaMike** [comment]
OK, I turned the system back on mid-afternoon yesterday & ran some command line diagnostics on the mega-bas.  The system appeared to be running "normally" when I ran them.  Results follow:

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/4aed8490-7432-4cd0-88b0-b0abaf64e45c)

Sometime around midnight the system freaked out again.  I ran the diagnostics again before powering  off the system.  Below are the results:

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/084c5511-163c-4e06-b540-3fa352348071)

So it does not appear to me that Nixie/REM are the issue.  I checked the wiring from the sensors to the Mega-Bas & all looked good.  I  removed & reseated the terminal block for the inputs.  I turned on the system this morning a couple of hours ago & all appeared normal again--but not for long!  The Polaris did not turn off as scheduled.  When I looks at dashPanel, the freeze protection was back on & (obviously) all of the sensor inputs (air & water temperature and flow switch in the flow cell) were zero/off again.

A soft reboot did not remedy the situation, when the Pi stack rebooted, the sensor inputs on the Mega-Bas were still zero.  I may try moving the terminal block from inputs 1, 2, & 3 to 5, 6, & 7 (one block over) & reconfiguring REM feeds.  Somehow, I doubt the behavior will change, It feels to me like the Mega-Bas is toast.

Do you agree?  Is there anything else should try before buying another board?

Thanks for your insights & help!

**Reply by FlaMike** [reply]
I moved the terminal block to inputs 5,6, & 7 and reconfigured REM, set the board jumpers for those inputs.  At the *moment* the pool is running normally--it's  only been about 15 minutes as I write this.  For how long, I  don't know...

**Reply by rstrouse** [reply]
I would re-install the firmware on the BAS.  Whatever you do make sure REM is shut down before you do this as the BAS does not shut itself down while it is trying to update the firmware.  The bad part is that it will give you an indication that all is fine and have you scratching your head later.

**Reply by FlaMike** [reply]
Will do & thanks for the REM shutdown tip!  Something tells me you know of *someone* who did that very head scratch :-o

**Reply by rstrouse** [reply]
I gave that guy a shave this very morning.   Spent hours trying to figure it out and had different results after each upgrade attempt.  Even then I didn't learn my lesson and did it again months later.

**Reply by FlaMike** [reply]
I know a  guy just like the one  you shaved this  morning--all too well, I fear!

A tech guy at Sequent created a new (beta?) version of the megabas firmware for me to test.  I pulled & installed it per your instructions.  It includes a "reset" command (you can find it with megabas -h) that he claimed would reset the inputs(??).  He said  to run it if the inputs freaked out.  They did so almost immediately when the system  came up.

So, I ran the reset command & it looks  like it did a *factory* reset on the board & blew away my entire  megabas configuration.  Not all at what I expected or was prepared for!!!  I went to a trusty backup on Nixie/REM et al & I'm back to the inputs  being zero again.  I've run out of time for the day, so I just shutdown the pool, flipped the breakers, & will go to Plan B, whatever that is tomorrow.  I just hope Plan B is not Plan 9 from Outer Space!

I remain flummoxed & depressed.  I finally had ORP going with the gold probe & BOOM the megabas or something blew up on me.  Oh, what a world!

I hope your weekend & forthcoming holiday are much more fun than mine is turning out to be...

**Reply by tagyoureit** [reply]
Are you saying that you ran the reset on the Sequent board and that somehow changed your REM configuration?  That shouldn't have happened; the Sequent board/software knows nothing about REM.

**Reply by FlaMike** [reply]
I am not sure I would express what happened quite as you did, but here is the best of my recollection--it was several hours  ago & I do not have screen shots.

I ran the new command, in my case "megabas 0 reset" from the command line via an ssh connection.  The Pi responded with [Reset].  I don't recall if Nixie & REM were running at the time since I had shut them down using a PM2 command in order to install & compile the megabas firmware previously as suggested by @rstrouse.  If they were off, I restarted them via PM2 and/or rebooted the Pi.

In any case,  once the system was up & running, I opened REM's URL in a browser.  As I recall, from the generic devices tab, the temperatures were zero, one of the problems I've been having throughout.  So I clicked on the I2C devices tab.  Instead of seeing the Multi-relay board, megabas, EZO ORP, pH, & pressure devices, a list of hex addresses from zero to ??? was presented with "unknown" in the name fields.  At  that point, I pretty much said WTF,  this looks like a factory reset of the entire board, not just a reset of the inputs as was presented to me.  Here is a quote from the email I got from the Sequent tech support guy:

> I just added the reset command, please get the repo and reinstall the command.
> This test is useful if you can make a reset when the inputs are zero and see if they come back to normal values. If they do I can 
> implement a failsafe mechanism in firmware.

What I saw was not at all what I was expecting--getting temperatures back to normal, for example.

Because my primary activity for the past few days has been trying to get the controller working again and the fact that I didn't have a compiled list of device configurations & screenshots. I decided to go back to a backup copy of the SD card from when the controller was  working in the recent past.  

Unfortunately, when the system rebooted, the air and  water temperatures and filter pressure fluctuated rapidly between the "real"  values and zero.  They then appeared to stabilize with reasonable values, but only for a couple of minutes & then everything went to zero where they remained.  Because the temperatures reported via the megabas were zero, Nixie/REM went into freeze protect mode and started the filter pump.  The flow switch reading was also "off" even though I could see the float position for the switch should have closed the reed switch.  Everything was as messed up as it's  been for the pastfew days starting last weekend.

I then shutdown the Pi via the command line & turned off  the breaker.  For the life of me, I do not understand what's happening.  The temperature, filter pressure, and flow switch readings have  been stable for at least 6 months until last week.  Since then they've been a mess.  There haven't been any hardware changes & the backup SD card is from time when the system was a stable.

I'm concerned about turning the system on & leaving it unattended lest it turn on the overflow valve repeatedly & drain the pool below the skimmers as it did last weekend.  At the same time, we're getting lots of thunderstorms & I may need to turn on the pump & open the drain valve to prevent flooding (the pool is at the same level as our house & could *potentially* cause problems if it overflows).  So, I feel I'm damned if I do & damned if I don't.

Please understand, I know you and the Sequent tech are trying to help but I don't know how to help you determine how diagnose the source of the problem, let alone how to resolve it.  I truly appreciate your help & concern as I have all along.

**Reply by rstrouse** [reply]
The first thing I would do is turn off the freeze protect.   I have several of these running and do not see the adc issues you are experiencing with the BAS it has to be defective.

**Comment by FlaMike** [comment]
Thanks for your suggestions!

I can certainly turn off freeze protect as it's not needed at the moment.
I also was going to go back to the 5-8 terminal block for the inputs as it
seemed to work for at least a few hours today.  (I went back to 1-4 when I
switched to the backup SD card this afternoon.)

I also mentioned to the Sequent tech I thought the board had failed.  He
seems not to want to replace it under warranty, but I can't spend my life
playing with a flakey board.

**Comment by FlaMike** [comment]
BTW, do you think Sequent's cost-reduced (cheapened) smart fan is worth
getting?  I know you modified REM recently to accommodate it.  So far the
Pi's temperature has been below 130F.  I have a case fan that runs when the
Pi is warmer than 90F.  I'm not sure if a CPU fan will keep things any
cooler.

Thanks!

**Reply by FlaMike** [reply]
Early this morning (around 1:00 CST) I received a note from the tech support guy I've been working with at Sequent.  He asked me to confirm the input voltage to the board.  

I use a rectifier bridge to convert the Pentair transformer's 24VAC to DC.  From the time I purchased & tested it, the bridge has output 25 VDC +/-.  The Pentair transformer actually provides about 27 VAC.  The Mega-Bas has always reported input voltage of about 34 VDC.   I've never understood why or been able to reconcile the difference.  Rectifier output is not as "clean" as a DC transformer, perhaps?

Anyway, I checked the voltage at the power terminal block with a DMM this morning & got 25.7 VDC.  I sent a note to Sequent confirming the power supplied to the board.

Sequent responded with a shipping confirmation for a replacement board, under warranty.  The tech requested that I sent the Mega-Bas I have now so they may determine what has gone wrong.

So, although this chapter of  my journey has been trying, I am hoping to close it  successfully within a week or so.  At the moment, my pool is not running, so I may program the pump & salt cell to operate manually on a schedule (sans controller) and control the acid pump with a WiFi outlet via HA.  I've done this previously when the ET boards kept blowing their RS-485 chips and I had to wait weeks for replacements.  

There seems to be something about pool equipment failures--they always happen in or about July for me.  (3 out of 4 crappy GloBrite lights bit the dust in the past 2 weeks.  I don't have a solution in hand yet & am still struggling to find one.)  I know I am very fortunate to have these 1st world problems, but they still suck.

**Reply by rstrouse** [reply]
That is very odd that the BAS is reporting 34vdc and is outside of the voltage parameters of the regulator on that board.   I wonder if all this is simply a bad linear regulator on the board.  The reason why the Pentair transformer is reporting 27vac and the and the bridge rectifier is reporting 25vdc is because only half of the AC current passes at any given time through a bridge rectifier and the diodes likely have a 2v drop over them.  Diodes always reduce voltage.

**Reply by FlaMike** [reply]
I can only hope the replacement does not share the same bad component!

**Reply by FlaMike** [reply]
According to USPS the replacement board will arrive on Monday.  In the interim, REM is dead & took Nixie with it.  I am *occasionally* able to power everything up, get good input readings for temperatures, flow, ORP, pH, & pressure.  Within a few minutes, things begin to collapse & everything goes to zero/no input.  Following by only a few moments, REM goes completely offline so Nixie is WTF am I supposed to do?  Pumps either turn off & the pool is dead in the water (so to speak).  Strangely, I can sometimes power cycle the world & normalcy returns, but only briefly & the death spiral returns.  Of course, the Pi doesn't crash & I'm guessing that REM has a seizure of some kind do to voltage management issues.

Fortunately, I'm able to get my backup plan in action.  The filter pump is running via an onboard schedule.  I wasn't able to get the salt cell to run without the Pi stack being powered up--no LEDs on the cell.  So, Plan B (chlorine tablets in a floater) is already deployed.  Not pretty, I will rely on the Taylor kit to tweak Cl and pH, but it's only for a week & I'm hopeful I can avoid an algae bloom and disease-infested water.  I can manage the overflow drain valve manually if we get torrential rains.  In the interim, I can focus on the non-pool challenges that life throws at me. 

Happy 4th of July to all!

**Reply by johnny2678** [reply]
so much fun!  wish I had any ideas to throw your way but I think you've already landed on the best path by waiting to try the new board.

I'm in a similar boat, while REM/njspc are working, I haven't had new ph/orp probe readings for 5 days while I wait for my new T3 board to arrive.  really hoping that solves my problem.

**Reply by FlaMike** [reply]
I hope the replacement board does the trick for you & lasts for many trouble-free years!  I feel like I'm back in the stone ages--no ORP, no pH, no salt, etc.  Not a single luxury. :-(

I have to hand it to Sequent.  Over the weekend the customer support rep was very responsive & tried his best to help out.  In the end, he agreed the board was a mess & sent a replacement gratis.  If he was extra nice, he included a return shipping label for returning the bad board, but I'll bite the bullet, if need be.

BTW, if anyone would like an A-S platinum industrial ORP probe used less than 6 months, I have one & am willing to deal.  As-is, but working as far as I can tell.  Caution: do not use it in a saltwater pool!

**Reply by rstrouse** [reply]
It will keep the chip cooler.  Active cooling on the chip is helpful at 140f the pi will begin to throttle.

**Reply by FlaMike** [reply]
I just ordered the Smart Fan.  It looks like it should arrive on the same day as  the replacement Mega-Bas.  So, I should only need to re-stack the Pi & HATs only once.  Whew!

Based on a quick reading of smart fan discussions, it looks like I need to do a pull for latest version of REM, assuming the fan I'll get will be version 6 (extra brain-dead & cost reduced).  Am I correct about needing to update REM?

Thanks & have a super holiday!!!

**Reply by rstrouse** [reply]
I believe so.  I do not have a v6 fan.  I relied on the goodwill of others.  They greatly simplified the fan in later versions.  I think when it first came out it was in the way too much stuff on the fan craze.  Led lights and screen display were the rage.

**Reply by FlaMike** [reply]
OK, I will plan on doing so once the new boards are in the stack.  I assume I'm getting V6 but who knows?  It's amazing how over-featured a fan can be.  I really don't care if it runs 24/7 at full speed.  It's not like we're going to get frost forming on the boards & shorting everything out.

Anyway, thank you for your ongoing help & support.

**Reply by johnny2678** [reply]
I've been running the v2.0 board for a while now.  Not sure what the spec difference is from v2 -> v6.  Hope they put a better fan on it tho.

Mine dropped out last summer but luckily it was easily replaced with a $12 Noctua fan from Amazon.  You should get some good mileage out of yours but just in case, I made a post about it [here](https://github.com/tagyoureit/nodejs-poolController/discussions/759
).

The money shot is the A/B test on the CPU temps before/after swapping the fan.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/0484fc42-7a1d-40da-9dbf-8fe78bf48c5a)

**Reply by FlaMike** [reply]
Thanks for your post & the link to your "fan with a seizure" post.  I had seen that post a while ago & was wondering if I should just get the Noctua & forget it.  So, that's what I'm doing--already ordered it from your link to Bezos World.  It's a pain restacking everything & I  shudder at the thought of losing  a teeny tiny screw/nut/whatever at the pad.  I put a Noctua in for my case fan to give it some longevity.

The replacement Mega-Bas, Smart Fan, & Noctua fan should all get here on Monday.  If so, the restacking (done away from the pad where itsy bitsy parts are less likleyto evaporate??) should be one a done-if all goes well!

BTW, where in Florida are you?  I'm in the Panhandle between Panama City Beach & Destin.  Lower Alabama, as it's called (aka Redneck Riviera).  Likely so far from you I'm in a different time zone (CST here).

**Reply by johnny2678** [reply]
Floribama!!  That's a beautiful part of the state.  It's been a bit since i've been up your way but I used to visit Panama City in a former life.

The closest I get these days is when I'm in Tallahassee for work.  I'm in Sarasota and that's still about a 5 hour drive for me!

Figured you guys would enjoy this...  I haven't even been working on my pool.  Since I lost my ph/orp readings I just cranked the SWG output to 45% and let it ride until my replacement T3 gets here.  Today, for fun, I just checked my levels and for some reason, right at 2:05 PM (ET), the both started sending readings again.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/2a24aef3-311d-49d8-8f4e-40d875544a98)

5 straight days of static readings and then it just started working out of the blue.  My mind is blown.  haha.  

Obviously, this is less than ideal, since I haven't been able to isolate the dropouts, and I will still probably sub in the new T3 when it gets here.

Just thought I'd drop that here so you knew you aren't the only one here with pool head-scratchers.  All the best. Happy 4th!

**Comment by FlaMike** [comment]
I love Sarasota.  Went to beautiful Crescent Beach from the 70s to the
90s.  The city has exploded since I first went there.  We went to Lido Key
for a week last November.  Where else do you go on vacation when you live
near the beach, but to another beach? 😎

Our first trip to the Panhandle was to Perdido & we could walk to the
Floribama.  It's a couple hours west from here but...

Small world, isn't it?!

Hope you have a wonderful 4th, too!

**Comment by FlaMike** [comment]
I just received the replacement Mega-Bas and installed it.  After tearing my few remaining hairs out, I *finally* got everything working again--as far as I can tell, anyway.  The biggest hurdle was controller the stinking filter pump.  Apparently, the replacement board did not see the configuration for RS-485 Pass Thru Mode or it must be set explicitly on the board.  Anyway, a couple of hot & frustrating hours later, I am back in business!  Yay!!

I am still awaiting a Smart Fan & Noctua to be used instead of the brand X Sequent provides, but both should be here in a few days.  Only one more Pad Ballet, I hope.

I did notice the Mega-Bas input voltage is reported as  33.05 volts.  Am I going to fry another Mega-Bas board, if that's  what happened to the old one?  My DMM shows the voltage at the Mega-Bas to be 25+/- VDC, as before.  Is this situation some  strange & possibly board-frying artifact of using a rectifier bridge instead of a transformer?  This is the bridge I am using:

https://www.amazon.com/gp/product/B01JIKSHCA/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

Should I get a different bridge & if so, which one?  Should  I just punt & get a 120 VAC to 24 VDC transformer, per the Wiki?  I just don't care to run into this issue again.  

One other strange happening in the resurrected system.  I have an Intellichlor 20 salt cell, configured as such in the dashPanel.  When the salt cell is running, it now shows dosing in ml of chlorine and it's  always zero.  The cell is on at 100% according to  its LEDs.  I believe the hardware tab is set correctly.  Not sure why this is  happening.  It may not be material, but it's  different & odd.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/b737de94-8089-468e-bd6b-cae4234c4091)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/a2f7bd8a-f306-48fa-8263-012d91640645)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/2c37341e-d14b-42d3-b714-b10682ea03cd)

Thanks for your help & suggestions!!!

**Reply by rstrouse** [reply]
A couple of things.

1. Measure the output from that rectifier in AC volts.  What does it say?
2. Set the chlorinator model in the chlorinator configuration.

**Reply by FlaMike** [reply]
AC into rectifier: 28.9 V
AC out from rectifier: 54.5 V
DC out from rectifier: 25.8 V

The salt cell was set correctly, Intellichlor 20 .  I changed it to Intellichlor 40, saved the chlorinator, changed it back & saved it again.  There was no change on the dashPanel, however.

**Reply by rstrouse** [reply]
I suspect that the BR is still outputting 33vac through its smoothing capacitor ~.6 x 54.5 = ~33v.  There is no regulator on the rectifier.  I recommend getting a transformer.

**Reply by FlaMike** [reply]
OK.  I will do so.  Thanks for your help!!!!

**Reply by FlaMike** [reply]
Do you think a Meanwell SDR-120 which provides 5A at 24 VDC will provide sufficient headroom for my setup?  I'm currently running a Pi 3B+ (2.5 A, @ 5 volts), a Building Automation Hat, 8 relay Hat, & Smart Fan in the Pi stack.  Plus A-S Whitebox hat with pH, ORP, and pressure sensors. I don't know how much power they require.  I also have 4 3-HP relays controlled by the stack--I have not found the power consumption for the 3 HP relays.  There's also 1 actuated valve, but it uses AC, of course.

It looks like the Pi stack itself is likely in the range of 3-3.5 A at most.  The 3 HP relays and A-S stuff are the big question marks, I think.  I also want to have a few spare amps if/when the Pi dies & I have to get a 4 or 5 or ??? to replace it.  I know they use more power.

Thanks!!!!!

**Reply by FlaMike** [reply]
Never mind.  I decided to get the "big boy" (SDR-240-24) and call it a day.  I've had enough challenges with my build & just want to back to my happy place (trying to swim laps in a 90+ degree pool without passing out from the heat!).

**Reply by rstrouse** [reply]
The relay coils do not draw much and are typically rated in the 2-5w range between inrush and hold.  This is in the .08A to .2A range.

**Reply by FlaMike** [reply]
Thank you for the additional information!  While the 5 amp transformer may be sufficient & provide some headroom for any additions to the pool or a more powerful Pi, I'm going  to stick with the overkill 10 amp unit & hope it proves to be the tonic that's needed and then some.

**Reply by FlaMike** [reply]
OK, yesterday I installed  the Meanwell transformer & powered the system back up.  I had it off while waiting for the transformer, lest the bridge rectifier be the source of the problems and to prevent it from blowing the new Mega-Bas.  The voltage at the transformer output was 24.0 DC on my DMM, spot on from the factory!  Yay!  The reading was taken before connecting power to the Pi stack (via the Mega-Bas).  At the Mega-Bas the input voltage reading was 21.99.  A little surprising that it was lower than the DMM, but hopefully better & less damaging than the 34 VDC readings at the Mega-Bas when the rectifier bridge was powering the board.  (At the bridge, the reading was 25-26 VDC).

Everything seemed to work as expected.  Yay again!  I ran the Polaris manually for a while, the system shut down at the appointed hour.  I figured I was in fat city again & was all set to play with the Smart Fan today.

Well, it was not to be.  In the middle of the night the temperature and voltage readings at the Mega-Base went to zero again & the system went into freeze protect mode.  I did hard & soft system resets to to avail.  My daemon had returned.

The system seems to oscillate rapidly between accurate temperatures and voltages and zero and then seems to settle at zero. The relays for the filter pump toggle on/off rapidly & the overflow drain relay turns on & more-or less stays on.  WTF???  Even the case fan relay becomes unstable (it should turn on when the Pi reports temperature over 90 F.)  I killed the system once again & am running the filter pump manually and using a chlorine floater with tablets to keep the chemistry somewhat stable.

To say the least, I am very dazed & confused, not to mention frustrated.  I switched to a backup SD card without success.  At this point I am not sure what to try next.  The system ran like a clock for 6 months before flipping out.

I have two theories and I hope one of them proves to be correct.  First, I wonder if an SD card became corrupted due to power fluctuations from our utility (very likely) and the corruption has  been replicated in my backups when cloning the card.  To test this theory, I guess I need to build everything from scratch starting with a fresh download of the OS.  Hoping I have enough screenshots of the Nixie & REM settings to minimize mistakes along the way.

The only other thing that comes to mind is the Pi is unstable.  Why?  Who knows--perhaps the same unstable power from the utility.  We do get all too frequent stuttering power.  Nearly all of my home automation stuff can handle it, but a few devices are unhappy when it occurs.  Perhaps a several year old 3B+ just goes goofy & I am better off getting a new one.

I don't understand why most of the time, the system seems OK during the day & then freaks out in the middle of the night.

If you have any other suggestions on what to do to fix my achy breaky Nixie, please let me know.  Thanks!

**Reply by johnny2678** [reply]
These might be elementary, but can you disable freeze protection? That would still allow your schedules to run and chem to dose, right?

What about replacing your 10k temp lead? Or at the least, stripping the wires back a little further for better contact with the mega-bas?

**Reply by FlaMike** [reply]
As you suggest (and I appreciate) I can disable freeze protection to run the pumps and chemical dosing (assuming I also disable the flow switch lockout).  I can also disconnect the overflow valve from the relay board so the pool doesn't empty as it tried to a few weeks ago.

The failures are not solely water temperature, however.  All of the inputs, air, water, flow switch, Pi CPU temp, Mega-Bas voltage input, etc., *all* get messed up during the "freak out sessions.:"   Something fundamental appears to fail and I can't isolate what it is.  

If only the air temp was  flaky, rewiring/replacing it makes sense.  I  don't understand why everything just  poops out in exactly the same way with the new Mega-Bas as previous one.  I checked all terminal block connections when I swapped the Hats, & everything seemed OK.

If I had any hair left, I'd certainly tear it out.  However, that field has been bare for decades.

**Reply by johnny2678** [reply]
Does the failure always happen overnight?  Are there any schedule jobs (cronjobs, or other) that run on the pi?  

My MEGA-BAS is HW rev1 with firmware 1.09.  Rock solid. Haven't touched it since i originally installed it.  what's yours?

Since Sequent already sent out a new board, i guess hardware failure is less likely, but maybe others on here with the same board rev as yours can chime in?

**Reply by FlaMike** [reply]
The failures have typically (always?  I'm not certain) occur overnight, however the time varies--from around 12:30-1:00 AM to around 5:00 AM today.  I am not running anything on the Pi other than njsPC, REM, dashPanel, & PM2.  Certainly, I've not created any cron jobs.  Good questions, though!

I will have to check on the firmware version on the Mega-Bas. The board is 4.2--it has DIP switches instead of jumpers.  When the original board I purchased showed up, I was surprised to see them--the documentation available at the time did not show DIP switches  & I had to learn via trial by fire how to  set them appropriately for thermisters & the flow switch.  I think I was a near firstie on the new board, it was purchased in December  2023.

I just checked an email I sent to Sequent a week ago with the "old" board I was using at the time.  It has version 2.03 firmware on it.  Although the board's silkscreen says 4.2, it's  diagnotic says 4.00.  Very different than your board.  I  assume the new board has the same versions but will have to confirm with you later--busy afternoon.

Like you, I'm interested in finding out if others with this board are having similar problems!!

I also need to play around with the Mega-Bas in failure mode again.  The original board I had showed the REM faithfully reported the input values seen by the Mega-Bas.  Perhaps I need to go  back to their tech support guy & see if this is a widespread problem.  If so, I'd rather get an older board Rev from them!!!

**Reply by FlaMike** [reply]
Interestingly, I replaced the SD card on the Pi with yet another fresh backup & restarted Nixie at al around 5:00 PM last night.  Everything appeared to be  working perfectly.  Good  reported air & water temperatures, flow switch reading was  good, etc.  The pool ran, ahem, swimmingly until it  shut down at 8:00 PM as scheduled.  It was delightfully quiescent overnight... Until 4:30 AM.  Then the daemons returned.

I was awakened by what I thought were pump noises.  None of my pool pumps should be running at that hour.  So I went to see what was  happening & lo and behold,  Nixie and REM reported air & water temperatures oscillated rapidly from around 81 degrees and zero and 90 degrees and zero, respectively.  Freeze protection went on/off on/off etc. and the filter pump was running amok as a result.  On the relay card, nearly all of the configured relays toggled like mad, with REM's indicator lights blinking like an old movie set computer.  Argh.

I tried to disable freeze protection, but Nixie was very slow to respond and it did not appear that I could regain control of things from a console.

I took a  few  quick screen shots, shut down the Pi via SSH and went outside to flip the breaker to enable the on board filter pump schedule to control water flow.  When I got to the pad, I saw  the heater was on, too!  Why I don't know--it's configured to be off in Nixie as I  don't need to warm the water this time of year in Florida.  I killed all of the breakers except for the filter pump & slowly walked  away shaking  my head.

I think I am gradually zeroing in the cause of the problems.  It seems to me the source of the problems is the Universal Inputs on the Mega-Bas.  They innocently appear to be normal from time to time, but then the evil spirits take over the reported values oscillate wildly and rapidly between the "correct"  values and zero.  Nixie & REM try to do the "right" thing, but the inputs are too unstable to keep up with.  The pool equipment gets dragged along for the ride.

I am very suspicious of the changes to Mega-Bas V 4.2 by Sequent.  They swapped out using jumpers to configure the Universal Inputs & other parts of the board and replaced them with DIP switches.  No problem there, but I  don't know what else they may have changed on the board.  More component cost cutting?  Erroneous firmware changes?  Manufacturing errors?  I don't know.  What I do know if the board seems to fail when the operating environment temperature is over 100 F, based on Pi CPU temperature readings.

I've sent a note detailing my latest experience & theories & asked them to replace my board with an earlier version with jumpers.  We'll see how they respond.  Fingers crossed!

Does anyone out there have version 4.2 of the Building Automation HAT?  Have you had any issues with it?  Please chime in, so we can determine if the boards are defective or something else is  going on.

Thanks!!!

**Comment by TopJim** [comment]
I can't speak to any of your issues specifically, but I had a lot of curious problems with my setup and concluded it was the SD card. It was a high-end Samsung SD card made for video cameras, so I don't think it was the SD card, per se, but the technology itself as SD cards were not meant to be read/written constantly. I switched to a SSD and it cleared up all my issues.
SSD: https://www.amazon.com/dp/B08K79T9G5?psc=1&ref=ppx_yo2ov_dt_b_product_details
Enclosure: https://www.amazon.com/dp/B0CCB5P51Y?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Reply by FlaMike** [reply]
Thank you for your response & suggestion!  I was using and "old school" 2-1/2" SSD and USB enclosure when the Nixie freaked out several weeks ago.  I have the same concerns regarding the longevity of micro SD cards.  I now have two Samsung 128 GB EVO Select micro SD cards that are clones.  One is in  the Pi and  the other in my desk, ready as  needed.  I clone them as needed (frequently, over the past few weeks) with Balena Etcher--it's quick & reliable.

I was concerned the old school SSD--a no brand beast--might have too high a power draw for the Pi.  As a result, when Nixie went nuts the first time, I switched back to the micro SD cards.  However, your suggestion gives me pause on going back to micro SDs.  It's a shame I just missed Prime Day.

Last week at the suggestion of Sequent's tech guy I double-checked all connections.  I use ferules & everything looked good.  However, I changed the male terminal block used to connect the thermisters and flow switch to the Mega-Bas to a brand new one.  I hesitate to say this, but knock on wood, Nixie has been working flawlessly since.  (I hope I didn't just jinx myself!)  Was the terminal block the problem?  I have no idea.

To help beat the heat in the enclosure, I also added a Smart Fan.  (I already had a case fan.)  Amazingly, now the Pi is almost always 20F above the ambient temperature, regardless of the time of day.  I don't have details on its previous temperature, but I don't believe the Pi has been warmer than 120F since the Smart Fan was installed.  Not too bad given the morning sun blasting the enclosure!

An NVMe SSD might be a reasonable migration.  Possibly a lighter power draw & certainly more compact.  I can always use the 2-1/2" old guy as a backup clone for it.

Thanks again for the suggestion!

**Reply by TopJim** [reply]
That Nvme enclosure I linked gets its power separate from the raspi, so no issues with being underpowered. I have mine hooked up to an active USB hub along with the keyboard, mouse, and SD reader/writer. I keep clone backups on the microSD cards that I no longer otherwise use (Samsung EVO), so having them on the usb hub makes it easier to insert/remove than on the Raspi itself. I’m not sure Raspi has enough current available if it’s plugged in to the Raspi directly.

**Reply by FlaMike** [reply]
Thanks for the additional information.  Mine would be just hanging out in the enclosure so I need to noodle how to provide separate power to it.  I just added an overpowered Meanwell transformer to the box, but it's 24VDC.  A buck converter and a butchered USB cable might do the trick...

**Reply by TopJim** [reply]
Or just hook it to the 5 vdc bus where your raspi is getting its power?

**Comment by FlaMike** [comment]
Doh!  Yep!

**Reply by FlaMike** [reply]
I did some quick & dirty calculations regarding power availability and power consumption should I just go back the the SSD it used prior to the freak outs.  Here is what I found:
    Mega-Bas can provide 5V and 3 amps to the Pi
    The Pi 3B+ that I'm using has a recommendation of using 5V and 2.5A supply
    USB A 2.0 can supply 5V and .5A
    The SATA drive I have uses between 0.06 and 0.4A

In addition, I have turned  off  WiFi on the Pi because I use Ethernet and the Pi's OS is a Lite version as it runs headless and I SSH to configure things as needed.  I'm guessing this configuration reduces the Pi's power consumption further.

Here's some data I found regarding various model's power consumption.  I don't think Nixie and REM tax the Pi nearly as much as video.

![image](https://github.com/user-attachments/assets/9a8af7f1-63a3-4448-8b6a-cf94fbf0042d)

If my quick & dirty calculations are correct, the Pi with an external SATA drive uses around 1.1A under fairly heavy load--far less than the Mega-Bas can provide.  Assuming the numbers I found are all accurate, it appears to me I can safely use the external SATA SSD and power it though the Pi's USB port without taxing the available power.  I guess this makes sense, since a Pi3B+ can be configured to boot from an SSD connected to its USB port.

If any of this seems off, please let me know!

**Reply by FlaMike** [reply]
@TopJim It looks like overall, NVMe drives use more power than SATA drives.  According to this  web site: https://driveshero.com/nvme-vs-sata-ssd-power-consumption/ it looks like a "typical" NVMe drive (whatever that means) uses about 1.6A and idle they use 0.7A.  

EIther type of drive could be powered via a Mega-Bas for most Pis, with plenty of headroom, I think.

Based on how you're powering your drive, it;s certainly not a problem.

**Reply by TopJim** [reply]
Yes, they run warm, too. The enclosure acts as a heat sink.

**Comment by FlaMike** [comment]
Ah yes, the hobgoblin of higher density & speed.  Fortunately, your
enclosure was selected because of its heat dissipation & neat display!
